### ISSUE TITLE
Transformers occasionally locking up in pub serve

### ISSUE THREAD

**Issue by blois** [angular/angular.dart#887]
The angular transformers are sharing a cached resolver state to have faster build times, but this cached resolver is locked while it's updating for async IO.

In pub serve, there are cases where a transformer in a later phase is being invoked before one in an earlier phase. This causes the IO for the first transformer to block waiting on the other transformer, but the second transformer cannot proceed because the first one has the resolver cache locked.

Avoiding the resolver cache completely increases total transformer time by 4x for pub build and 160x for pub serve incremental updates.

I'm investigating a couple of approaches which hopefully will not impact build or update times.
